packages/edit-site/src/components/page-pages/index.js

```diff
@@ -131,12 +153,20 @@ const DEFAULT_STATUSES = 'draft,future,pending,private,publish'; // All but 'tra
 export default function PagePages() {
 	const postType = 'page';
 	const [ view, setView ] = useView( postType );
-	const [ pageId, setPageId ] = useState( null );
 	const history = useHistory();
+	const { params } = useLocation();
+	const { isCustom = 'false' } = params;
 
 	const onSelectionChange = useCallback(
-		( items ) => setPageId( items?.length === 1 ? items[ 0 ].id : null ),
-		[ setPageId ]
+		( items ) => {
+			if ( isCustom === 'false' && view?.type === LAYOUT_LIST ) {
+				history.push( {
+					...params,
+					postId: items.length === 1 ? items[ 0 ].id : undefined,
```

Comment:
The selection change persists in the url which allow us to update the "preview" area with the right postId.

Reply:
This is nice. As a follow-up, we should consider updating the content area as well (update the selection).

Reply:
you mean making the selection in dataviews controlled right?